osgi/impl/src/test/java/org/pentaho/ctools/cde/environment/CdeEnvironmentTest.java

```diff
@@ -0,0 +1,223 @@
+ */
+
+package org.pentaho.ctools.cde.environment;
+
+import org.apache.commons.lang.StringUtils;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mockito;
+import org.pentaho.ctools.cde.datasources.manager.DataSourceManager;
+import org.pentaho.ctools.cde.plugin.resource.PluginResourceLocationManager;
+import pt.webdetails.cdf.dd.IPluginResourceLocationManager;
+import pt.webdetails.cdf.dd.datasources.IDataSourceManager;
+import pt.webdetails.cdf.dd.model.core.writer.IThingWriterFactory;
+import pt.webdetails.cdf.dd.model.inst.Dashboard;
+import pt.webdetails.cdf.dd.model.inst.writer.cdfrunjs.amd.CdfRunJsThingWriterFactory;
+import pt.webdetails.cdf.dd.model.inst.writer.cdfrunjs.dashboard.CdfRunJsDashboardWriteContext;
+import pt.webdetails.cdf.dd.model.inst.writer.cdfrunjs.dashboard.CdfRunJsDashboardWriteOptions;
+import pt.webdetails.cdf.dd.structure.DashboardWcdfDescriptor;
+import pt.webdetails.cpf.repository.api.IBasicFile;
+import pt.webdetails.cpf.repository.api.IContentAccessFactory;
+import pt.webdetails.cpf.repository.api.IReadAccess;
+import pt.webdetails.cpf.repository.api.IUserContentAccess;
+
+import static org.hamcrest.CoreMatchers.instanceOf;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertThat;
+import static org.mockito.Matchers.anyObject;
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Mockito.when;
+
+public class CdeEnvironmentTest {
+  private static final String PLUGIN_REPOSITORY_DIR = "/public/cde";
+  private CdeEnvironment cdeEnvironment;
+  private DataSourceManager dataSourceManager;
+  private PluginResourceLocationManager pluginResourceLocationManager;
+
+  @Before
+  public void setUp() {
+    cdeEnvironment = new CdeEnvironment();
+    dataSourceManager = Mockito.mock( DataSourceManager.class );
+    cdeEnvironment.setDataSourceManager( dataSourceManager );
+    pluginResourceLocationManager = Mockito.mock( PluginResourceLocationManager.class );
+  }
+
+  @After
+  public void tearDown() {
+    cdeEnvironment = null;
+  }
+
+  @Test
+  public void testGetLocale() {
+    assertNull( cdeEnvironment.getLocale() );
+  }
+
+  @Test
+  public void testGetResourceLoader() {
+    assertNull( cdeEnvironment.getResourceLoader() );
+  }
+
+  @Test
+  public void testGetCdfIncludes() throws Exception {
+    assertNull( cdeEnvironment.getCdfIncludes( null, null, false, false, null, null ) );
+  }
+
+  @Test
+  public void testGetExtApi() {
+    assertNull( cdeEnvironment.getExtApi() );
+  }
+
+  @Test
+  public void testGetFileHandler() {
+    assertNull( cdeEnvironment.getFileHandler() );
+  }
+
+  @Test
+  public void testGetUrlProvider() {
+    assertNull( cdeEnvironment.getUrlProvider() );
+  }
+
+  @Test
+  public void testGetUserSession() {
+    assertNull( cdeEnvironment.getUserSession() );
+  }
+
+  @Test
+  public void testRefresh() {
+    cdeEnvironment.refresh();
```

Comment:
Is this testing an error? If so, it should be explicit in the test declaration

Reply:
its testing a NOOP dummy function required by the current core implementation, I'll remove the useless test